### PR TITLE
[DUOS-2907, DUOS-2917][risk=no] Bug fix: Update use effect handler

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
* https://broadworkbench.atlassian.net/browse/DUOS-2907
* https://broadworkbench.atlassian.net/browse/DUOS-2917

### Summary
* Credit to @fboulnois for the fix
* Updates the use effect handler to eliminate an infinite reload problem

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/DatasetSearch.js [hunk 1]
```diff
@@ -194,26 +195,30 @@ export const DatasetSearch = (props) => {
...
-        await DataSet.searchDatasetIndex(query).then((datasets) => {
-          setDatasets(datasets);
-          setLoading(false);
-        });
-      } catch (error) {
-        Notifications.showError({ text: 'Failed to load Elasticsearch index' });
+      if (!loaded) {
+        if (institutionSet) {
+          Notifications.showError({ text: 'You must set an institution in your profile to view the `myinstitution` data library' });
+          //props.history.push('/profile');
```
Comment (fboulnois) [inline src/pages/DatasetSearch.js]:
```suggestion
          props.history.push('/profile');
```

We should uncomment this line and ensure that `props` is passed in too.

**Review by fboulnois — CHANGES_REQUESTED** [review]
One change:

**Review by aarohinadkarni — APPROVED** [review]
awesome 👍

File: src/pages/DatasetSearch.js [hunk 2]
```diff
@@ -194,26 +195,30 @@ export const DatasetSearch = (props) => {
...
+        } catch (error) {
+          Notifications.showError({ text: 'Failed to load Elasticsearch index' });
+        }
+        setLoaded(true);
       }
     };
     init();
-  }, [institutionId, isInstitutionQuery, isSigningOfficial, props.history, version.query]);
-
+  }, [loaded, institutionSet, fullQuery, props.history]);
```
Comment (fboulnois) [inline src/pages/DatasetSearch.js]: Here's an updated solution that avoids using an extra state variable:

```
  const fullQuery = assembleFullQuery(isSigningOfficial, isInstitutionQuery, version.query);
  const isInstitutionSet = institutionId === undefined && isInstitutionQuery;

  useEffect(() => {
    const init = async () => {
      if (loading) {
        if (isInstitutionSet) {
          Notifications.showError({ text: 'You must set an institution in your profile to view the `myinstitution` data library' });
          props.history.push('/profile');
          return;
        }
        try {
          await DataSet.searchDatasetIndex(fullQuery).then((datasets) => {
            setDatasets(datasets);
            setLoading(false);
          });
        } catch (error) {
          Notifications.showError({ text: 'Failed to load Elasticsearch index' });
        }
      }
    };
    init();
  }, [loading, isInstitutionSet, fullQuery, props.history]);
```

**Review by fboulnois — COMMENTED** [review]
See below:

**Review by fboulnois — APPROVED** [review]
👍

**Pull request merged**